ambari-server/src/main/java/org/apache/ambari/server/topology/TopologyRequestUtil.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ambari.server.topology;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+import static java.util.stream.Collectors.toSet;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.ambari.server.state.StackId;
+import org.apache.ambari.server.utils.JsonUtils;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+
+/**
+ * Utility functions for topology requests.
+ */
+public class TopologyRequestUtil {
```

Comment:
Rather than having this utilities class, perhaps we should consider building out a class that can represent the cluster creation template, and that class could then handle the responsibilities of serializing, de-serializing, filtering out passwords, etc.

Reply:
Attila has created the BlueprintBasedClusterProvisionRequest class to encapsulate all information coming from the blueprint and cluster template. Persisted stackId's will land here.

Reply:
The cluster creation request is represented by `ProvisionClusterRequest`.  I think we should consider moving these utility functions there, or to `TopologyRequest` itself, depending on whether also apply to scale request, too, or only provisioning.